src/lib/Libifl/pbs_loadconf.c

```diff
@@ -275,22 +275,27 @@ parse_psi(char *conf_value)
 	char **list;
 	int i;
 	char *svrname = NULL;
+	char *local_conf = NULL;
 
 	free(pbs_conf.psi);
 	free(pbs_conf.psi_str);
 
-	if (conf_value == NULL)
-		return -1;
+	if (conf_value == NULL) {
+		pbs_asprintf(&local_conf, "%s:%d", pbs_default(), pbs_conf.batch_service_port);
+		if (local_conf == NULL)
+			return -1;
+	} else
+		local_conf = conf_value;
 
-	list = break_comma_list(conf_value);
+	list = break_comma_list(local_conf);
 	if (list == NULL)
 		return -1;
```

Comment:
I think u need to free local_conf from here and before all returns also, may use `goto` and use condition like
```c
if (conf_value == NULL)
    free(local_conf)
```